pkg/webhook/server/generic/pod/pod_validating_handler.go

```diff
@@ -46,15 +47,19 @@ func NewValidatingHandler() *ValidatingHandler {
 }
 
 func (h *ValidatingHandler) Handle(ctx context.Context, req admission.Request) (resp admission.Response) {
-	if req.Operation == admissionv1.Delete {
-		return admission.Allowed("pod is allowed by opslifecycle")
-	}
-
 	logger := h.Logger.WithValues(
 		"op", req.Operation,
 		"pod", commonutils.AdmissionRequestObjectKeyString(req),
 	)
 
+	if req.Operation == admissionv1.Delete {
+		if err := gracedelete.New().Validating(ctx, h.Client, req); err != nil {
```

Comment:
I think you can move gracedelete to `webhooks` list and run together later in line 77.

Of course, the for loop statement need to be changed to remove the short-circuit logic and aggregate all the returned errors, in order to execute every validating hook.